contracts-core/package.json

```diff
@@ -10,6 +10,7 @@
     "@commitlint/config-conventional": "^9.1.2",
     "@ethersproject/abstract-signer": "^5.5.0",
     "@ethersproject/bignumber": "^5.5.0",
+    "@ethersproject/bytes": "5.0.0",
```

Comment:
What's this for? Seems surprising we'd need to add a dep in `contracts-core`

Reply:
It's a new warning that popped up during the `yarn install` process after upgrading to new node version. Thought this might be causing the failure. 
https://github.com/ScopeLift/umbra-protocol/actions/runs/5649918220/job/15306143560
```
warning "workspace-aggregator-d28f0833-fbcf-481a-914d-96f0486c469d > @umbra/contracts-core > @typechain/ethers-v5@10.2.0" has unmet peer dependency "@ethersproject/bytes@^5.0.0".
```

But note that even thought this PR is now passing, the CI has become very finicky where sometimes the `typechain` in `contracts-core` doesn't get generated before/when `umbra-js` tests are being run. Ben is saying it could be a race condition issue. Might be helpful to have you or @wildmolasses take a closer look.

Reply:
if `contracts-core` is specified as a dependency in `umbra-js`, then it will be built before `umbra-js` tests run. if it's not specified as a dependency, then lerna / yarn doesn't know that the two commands (building in contracts-core, tests in `umbra-js`) can't run parallel, or that there should be some order to them. I recommend adding contracts-core as a dependency. That then creates an issue for the standalone `umbra-js` package -- I don't think we've published `contracts-core` to our @umbracash namespace, so the package would be unavailable. Perhaps adding `contracts-core` as a devDependency, then, would be appropriate. Let me know if you want to sync on this in chat!

Reply:
Removed the `@ethersproject/bytes` package in https://github.com/ScopeLift/umbra-protocol/pull/577/commits/440e0deb48a49b4aab735fedeb007ef5ca19d915

Reply:
And solved the underlying failure by updating `hardhat` version in https://github.com/ScopeLift/umbra-protocol/pull/577/commits/e9b5b9b4369d1597e7c2916e6f7e792d3e460be4